desc/grid.py

```diff
@@ -662,23 +773,24 @@ def change_resolution(self, L, M, N, NFP=None):
             Number of field periods.
 
         """
-        self._NFP = NFP if NFP is not None else self.NFP
-        if L != self.L or M != self.M or N != self.N:
-            self._L = L
-            self._M = M
-            self._N = N
+        if NFP is None:
+            NFP = self.NFP
+        if L != self.L or M != self.M or N != self.N or NFP != self.NFP:
             self._nodes, self._spacing = self._create_nodes(
-                L=L,
-                M=M,
-                N=N,
-                NFP=self.NFP,
-                axis=len(self.axis) > 0,
-                endpoint=self.endpoint,
+                L=L, M=M, N=N, NFP=NFP, axis=self.axis.size > 0, endpoint=self.endpoint
             )
             # symmetry handled in create_nodes()
             self._sort_nodes()
-            self._find_axis()
-            self._scale_weights()
```

Comment:
Previously, when the grid changed resolution, none of the unique and inverse idx would be updated.